dotnet/samples/KernelSyntaxExamples/Example03_Variables.cs

```diff
@@ -21,10 +20,11 @@ public static async Task RunAsync()
         Kernel kernel = new KernelBuilder().WithLoggerFactory(s_loggerFactory).Build();
         var textPlugin = kernel.ImportPluginFromObject<StaticTextPlugin>();
 
-        var variables = new ContextVariables("Today is: ");
-        variables.Set("day", DateTimeOffset.Now.ToString("dddd", CultureInfo.CurrentCulture));
+        var arguments = new KernelFunctionArguments();
+        arguments["input"] = "Today is: ";
+        arguments["day"] = DateTimeOffset.Now.ToString("dddd", CultureInfo.CurrentCulture);
 
-        var result = await kernel.InvokeAsync(textPlugin["AppendDay"], variables);
+        var result = await kernel.InvokeAsync(textPlugin["AppendDay"], arguments);
```

Comment:
Just to confirm, syntactically this could also be written as:
```C#
var result = await kernel.InvokeAsync(textPlugin["AppendDay"], new()
{
    { "input", "Today is: " },
    { "day", DateTimeOffset.Now.ToString("dddd", CultureInfo.CurrentCulture) }
});
```
and it would compile and work, right?  Just want to make sure our design of KernelFunctionArguments doesn't wall that off, as it's relatively natural and will become even more so once collection expressions support dictionaries in C# 13.

Reply:
It should be possible; however, there is ambiguity: "The call is ambiguous between the following methods or properties: 
'KernelExtensions.InvokeAsync(Kernel, KernelFunction, string, CancellationToken)' and 'KernelExtensions.InvokeAsync(Kernel, KernelFunction, KernelFunctionArguments?, CancellationToken)'."

It can easily be fixed by specifying parameter name:

    var result = await kernel.InvokeAsync(textPlugin["AppendDay"], arguments: new()
    {
        ["input"] = "Today is: ",
        [ "day"] = DateTimeOffset.Now.ToString("dddd", CultureInfo.CurrentCulture),
    })

Please let me know if this solution is acceptable. If not, I will try to address it within the scope of the issue - https://github.com/microsoft/semantic-kernel/issues/3761